File: ems-cloud-mgmt-sdk/javascript-closure-angular/API/Client/OrderHistory.js
goog.provide('API.Client.OrderHistory');

/**
 * @record
 */
API.Client.OrderHistory = function() {}

/**
 * Apikey
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.apikey;

/**
 * Exchange id
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.exchangeId;

/**
 * Client order id
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.clientOrderId;

/**
 * Symbol id exchange
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.symbolIdExchange;

/**
 * Symbol id in coinapi
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.symbolIdCoinapi;

/**
 * Amount
 * @type {!number}
 * @export
 */
API.Client.OrderHistory.prototype.amountOrder;

/**
 * Price
 * @type {!number}
 * @export
 */
API.Client.OrderHistory.prototype.price;

/**
 * 1-buy, 2-sell
 * @type {!number}
 * @export
 */
API.Client.OrderHistory.prototype.side;

/**
 * Order type
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.orderType;

/**
 * Time in force
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.timeInForce;

/**
 * Expire time
 * @type {!API.Client.date}
 * @export
 */
API.Client.OrderHistory.prototype.expireTime;

/**
 * Exec inst
 * @type {!Array<!string>}
 * @export
 */
API.Client.OrderHistory.prototype.execInst;

/**
 * Client order id format
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.clientOrderIdFormatExchange;

/**
 * Exchange order id
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.exchangeOrderId;

/**
 * Amount open
 * @type {!number}
 * @export
 */
API.Client.OrderHistory.prototype.amountOpen;

/**
 * Amount filled
 * @type {!number}
 * @export
 */
API.Client.OrderHistory.prototype.amountFilled;

/**
 * Average price
 * @type {!number}
 * @export
 */
API.Client.OrderHistory.prototype.avgPx;

/**
 * Status
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.status;

/**
 * History status
 * @type {!Array<!string>}
 * @export
 */
API.Client.OrderHistory.prototype.statusHistoryStatus;

/**
 * History status time
 * @type {!Array<!API.Client.date>}
 * @export
 */
API.Client.OrderHistory.prototype.statusHistoryTime;

/**
 * Error message
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.errorMessageResult;

/**
 * Error message reason
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.errorMessageReason;

/**
 * Error message
 * @type {!string}
 * @export
 */
API.Client.OrderHistory.prototype.errorMessageMessage;

/**
 * Fills time
 * @type {!Array<!API.Client.date>}
 * @export
 */
API.Client.OrderHistory.prototype.fillsTime;

/**
 * Fills price
 * @type {!Array<!number>}
 * @export
 */
API.Client.OrderHistory.prototype.fillsPrice;

/**
 * Fills amount
 * @type {!Array<!number>}
 * @export
 */
API.Client.OrderHistory.prototype.fillsAmount;

/**
 * Created time
 * @type {!API.Client.date}
 * @export
 */
API.Client.OrderHistory.prototype.createdTime;

File: chain-sdk/clojure/src/ems_rest_api/specs/order_history.clj
(ns ems-rest-api.specs.order-history
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def order-history-data
  {
   (ds/opt :apikey) string?
   (ds/opt :exchangeId) string?
   (ds/opt :clientOrderId) string?
   (ds/opt :symbolIdExchange) string?
   (ds/opt :symbolIdCoinapi) string?
   (ds/opt :amountOrder) float?
   (ds/opt :price) float?
   (ds/opt :side) float?
   (ds/opt :orderType) string?
   (ds/opt :timeInForce) string?
   (ds/opt :expireTime) inst?
   (ds/opt :execInst) (s/coll-of string?)
   (ds/opt :clientOrderIdFormatExchange) string?
   (ds/opt :exchangeOrderId) string?
   (ds/opt :amountOpen) float?
   (ds/opt :amountFilled) float?
   (ds/opt :avgPx) float?
   (ds/opt :status) string?
   (ds/opt :statusHistoryStatus) (s/coll-of string?)
   (ds/opt :statusHistoryTime) (s/coll-of inst?)
   (ds/opt :errorMessageResult) string?
   (ds/opt :errorMessageReason) string?
   (ds/opt :errorMessageMessage) string?
   (ds/opt :fillsTime) (s/coll-of inst?)
   (ds/opt :fillsPrice) (s/coll-of float?)
   (ds/opt :fillsAmount) (s/coll-of float?)
   (ds/opt :createdTime) inst?
   })

(def order-history-spec
  (ds/spec
    {:name ::order-history
     :spec order-history-data}))
